### PR TITLE
Add Filtering to Leaderboard

### LINKED ISSUE #400: Add a way to filter in the leaderboard controller
Provide regex via query param to filter list and add meta data about total number of pages

### PR DESCRIPTION
**Pull request by 1-alex98**
Fixes #400

### CODE REVIEW

**Review by Brutus5000 — DISMISSED** [review]
I will perform some tests on test server before merging

File: src/main/java/com/faforever/api/data/domain/GlobalRatingWithRank.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.yahoo.elide.annotation.Include;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "global_rating_rank_view")
+@Include(rootLevel = true, type = "globalRatingWithRank")
+public class GlobalRatingWithRank extends RatingWithRank {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/data/domain/GlobalRatingWithRank.java]: shouldn't we add some read only permission?

**Comment by bukajsytlos** [comment]
also integration tests should be possible right?

**Comment by Brutus5000** [comment]
Actually there is no sense in keeping the old rating and having 2 rating entities

File: src/main/java/com/faforever/api/data/domain/GlobalRatingWithRank.java (hunk 1 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/data/domain/GlobalRatingWithRank.java]: No by default its read for everybody

**Comment by Brutus5000** [comment]
@bukajsytlos I would add integration tests only if there are permissions or other logic involved.

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**